tlsutil/config.go

```diff
@@ -535,7 +553,7 @@ func (c *Configurator) outgoingRPCTLSDisabled() bool {
 	}
 
 	// if CAs are provided or VerifyOutgoing is set, use TLS
-	if c.caPool != nil || c.base.VerifyOutgoing {
+	if c.base.VerifyOutgoing {
```

Comment:
This is an important change! Before providing a CA was enough to turn on TLS. But thats not actually desired, because it breaks the upgrade scenario, when one server is configured with a CA and uses TLS for outbound connections.

It shouldn't do that until `VerifyOutgoing` is turned on.

Until now that was not a problem, because we were using hardcoded values to make sure it behaves this way, but now this problem surfaced. Which is why it is fixed here.

Reply:
👍